src/main/java/commands/Command.java

```diff
@@ -0,0 +1,114 @@
+package commands;
+
+import exceptions.MizzException;
+import util.Storage;
+import util.TaskList;
+import util.Ui;
+
+/**
+ * Abstract class encapsulating a possible command.
+ */
+public abstract class Command {
+    protected CommandTypes command;
+
+    /**
+     * Enum represnting all valid commands. <br>
+     * Contains useful methods to interact with valid types.
+     */
+    public static enum CommandTypes {
+        TODO, DEADLINE, EVENT, LIST, MARK, UNMARK, DELETE, BYE, FIND;
```

Comment:
I like how you incorporated enums in your code here to represent the different commands